VideoLocker/src/main/java/org/edx/mobile/model/course/BlockModel.java

```diff
@@ -51,6 +51,6 @@
     public BlockData data;
 
     public boolean isContainer(){
-        return descendants != null && descendants.size() > 0;
+        return !type.isUnit;
```

Comment:
I guess we could add a null check here just for safety's sake, and fall back to the original logic:

````java
return type != null ? !type.isUnit : (descendants != null && descendants.size() > 0)
````

The `descendants.size() > 0` can be removed depending on whether we take an empty `descendants` array as a sign of it being a container or a leaf.